### PR TITLE
fix(southpark): change pooling strategy and fix bug

### LINKED ISSUE #145: examples/southpark-search `encode.yml` is not customizable
i guess `model_name` should be `pretrained_model_name_or_path` in `encode.yml`

### LINKED ISSUE #146: examples/southpark-search throws exception on other transformers model
model_name:- `distilbert-base-cased`
traceback:- https://pastebin.com/bmrpyHpL

### PR DESCRIPTION
**Pull request by YueLiu-jina**
resolve #145 
resolve #146

### CODE REVIEW

File: southpark-search/app.py [hunk 1]
```diff
@@ -24,9 +24,9 @@ def print_topk(resp, word):
             score = match.score.value
             if score < 0.0:
                 continue
-            doc = match.meta_info.decode()
-            name, line = doc.split('[SEP]', maxsplit=1)
-            print('> {:>2d}({:.2f}). {} said, "{}"'.format(idx, score, name.upper(), line.strip()))
+            character = match.meta_info.decode()
+            sentence = match.chunks[0].text
```
Comment (JoanFM) [inline southpark-search/app.py:28]: @nan-wang I think here we are having the same issue as in `urbandict`

**Comment by JoanFM** [comment]
LGTM!

**Comment by YueLiu-jina** [comment]
Here's the `resp.search.docs`, seems working well except the ugly chunks[0]
```
id: 1
weight: 1.0
length: 8
matches {
  id: 37
  chunks {
    id: 8577959
    weight: 1.0
    length: 1
    mime_type: "text/plain"
    text: "What the?"
    level_depth: 1
    parent_id: 37
    location: 0
    location: 10
    embedding {
    }
  }
  weight: 1.0
  length: 8
  meta_info: "Nathan"
  mime_type: "text/plain"
  score {
    value: 17.838159561157227
    op_name: "NumpyIndexer"
    ref_id: 1
  }
}
matches {
  id: 29
  chunks {
    id: 4010709927
    weight: 1.0
    length: 1
    mime_type: "text/plain"
    text: "Let\'s take a hot shower!"
    level_depth: 1
    parent_id: 29
    location: 0
    location: 25
    embedding {
    }
  }
  weight: 1.0
  length: 8
  meta_info: "KKKLeader"
  mime_type: "text/plain"
  score {
    value: 19.105012893676758
    op_name: "NumpyIndexer"
    ref_id: 1
  }
}
matches {
  id: 30
  chunks {
    id: 3246871765
    weight: 1.0
    length: 1
    mime_type: "text/plain"
    text: "Right!"
    level_depth: 1
    parent_id: 30
    location: 0
    location: 7
    embedding {
    }
  }
  weight: 1.0
  length: 8
  meta_info: "Protesters"
  mime_type: "text/plain"
  score {
    value: 19.645029067993164
    op_name: "NumpyIndexer"
    ref_id: 1
  }
}
matches {
  id: 24
  chunks {
    id: 1756328814
    weight: 1.0
    length: 1
    mime_type: "text/plain"
    text: "Yeh-hah."
    level_depth: 1
    parent_id: 24
    location: 0
    location: 9
    embedding {
    }
  }
  weight: 1.0
  length: 8
  meta_info: "Kyle"
  mime_type: "text/plain"
  score {
    value: 19.846115112304688
    op_name: "NumpyIndexer"
    ref_id: 1
  }
}
matches {
  id: 47
  chunks {
    id: 1979282394
    weight: 1.0
    length: 1
    mime_type: "text/plain"
    text: "You wanna see \'em?"
    level_depth: 1
    parent_id: 47
    location: 0
    location: 19
    embedding {
    }
  }
  weight: 1.0
  length: 8
  meta_info: "Cartman"
  mime_type: "text/plain"
  score {
    value: 19.934221267700195
    op_name: "NumpyIndexer"
    ref_id: 1
  }
}
mime_type: "text/plain"
text: "Randy[SEP]That\'s it, nobody\'s gonna help us!\n"
```

**Comment by YueLiu-jina** [comment]
This does not seem to share the same problem with urbandict. In southpark, one `matches` has only one `chunks`, and the text in `chunks` is the whole thing, so it looks not so bad. But urbandict cuts the definition into different chunks, i.e. one matches has 5 chunks, which is the most serious problem

**Comment by JoanFM** [comment]
> This does not seem to share the same problem with urbandict. In southpark, one `matches` has only one `chunks`, and the text in `chunks` is the whole thing, so it at least looks good. But urbandict cuts the definition into different chunks, i.e. one matches has 5 chunks, which is the most serious problem

I will take a closer look at it

**Pull request merged**